system/system_monitor/src/hdd_monitor/hdd_monitor.cpp

```diff
@@ -240,27 +236,28 @@ void HDDMonitor::checkUsage(diagnostic_updater::DiagnosticStatusWrapper & stat)
 
 void HDDMonitor::getTempParams(void)
 {
-  XmlRpc::XmlRpcValue params;
-
-  pnh_.getParam("disks", params);
-  if (params.getType() != XmlRpc::XmlRpcValue::TypeArray) return;
-
-  for (int i = 0; i < params.size(); ++i) {
-    std::string name;
-    TempParam param;
-
-    // Skip no name
-    if (!params[i]["name"].valid()) continue;
-
-    if (params[i]["name"].getType() == XmlRpc::XmlRpcValue::TypeString)
-      name = static_cast<std::string>(params[i]["name"]);
-
-    if (params[i]["temp_warn"].getType() == XmlRpc::XmlRpcValue::TypeDouble)
-      param.temp_warn_ = static_cast<double>(params[i]["temp_warn"]);
-
-    if (params[i]["temp_error"].getType() == XmlRpc::XmlRpcValue::TypeDouble)
-      param.temp_error_ = static_cast<double>(params[i]["temp_error"]);
-
-    temp_params_[name] = param;
-  }
+    // TODO(yunus.caliskan): Figure out the XmlRpc dependency.
+//  XmlRpc::XmlRpcValue params;
```

Comment:
@mitsudome-r @esteve Do you know how to treat this dependency? I assume it is from this package: http://wiki.ros.org/xmlrpcpp which is not used and available in ROS2. Since i am not sure what information is received and from where in this function, I am not clear on providing a substitute.

@jilaada CCing you because I noticed you are working on `autoware_auto_state_monitor` which also has this dependency.

Reply:
@cvasfi Take a look at this: https://github.com/tier4/AutowareArchitectureProposal/issues/84#issuecomment-713882022. I'm using this as reference, although the configuration files are slightly more conplex here and in `autoware_state_monitor` as they deal with multiple types.